packages/core/src/pubsub/pubsub.ts

```diff
@@ -66,6 +69,14 @@ export class Pubsub extends Observable<PubsubMessage> {
     // Textually, `this.peerId$` appears after it is called.
     // Really, subscription is lazy, so `this.peerId$` is populated before the actual subscription act.
     this.peerId$ = from<Promise<string>>(this.ipfs.id().then((_) => _.id))
+    this.pubsubKeepaliveInterval = setInterval(
+      this.publishPubsubKeepaliveIfNeeded.bind(this),
+      this.maxPubsubPublishInterval / 2
```

Comment:
Either we set the interval, then have to do math to figure out what the max delay between messages becomes based on that interval.  Or we set the max delay between messages, and do math to set the interval.  Since the thing we actually care about is how long it is between messages in the worst case, that seemed like the thing that made sense to set, and then work out the interval timing from that.

Reply:
I think the key thing I was missing here was in effect "we are doing this to reduce the amount of pubsub traffic to only what is necessary as much as possible". May be worth including in the existing comments.